jena-fuseki-war on tomcat - java.lang.NoClassDefFoundError: jakarta/faces/context/FacesContext

### Version

5.4.0

### What happened?

Hi there,
I'm using the jena-fuseki-war, deployed on tomcat. Older versions of the .war deploy just fine, but starting with 5.4.0, I'm getting `java.lang.NoClassDefFoundError: jakarta/faces/context/FacesContext` Errors (more detail below).
Is there some additional config required to run newer versions on tomcat?

Thanks

### Relevant output and stacktrace

```shell
java.lang.IllegalStateException: Error starting child
		at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:569)
		at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:652)
		at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1087)
		at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1853)
		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
		at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
		at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
		at java.base/java.util.concurrent.AbstractExecutorService.submit(Unknown Source)
		at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:998)
		at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:383)
		at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1582)
		at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:265)
		at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:109)
		at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:389)
		at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:336)
		at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:743)
		at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:769)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1164)
		at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1160)
		at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
		at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
		at java.base/java.util.concurrent.AbstractExecutorService.submit(Unknown Source)
		at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:716)
		at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:203)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.core.StandardService.startInternal(StandardService.java:412)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:866)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		at org.apache.catalina.startup.Catalina.start(Catalina.java:759)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
		at java.base/java.lang.reflect.Method.invoke(Unknown Source)
		at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:342)
		at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:473)
	Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/fuseki]]
		at org.apache.catalina.util.LifecycleBase.handleSubClassException(LifecycleBase.java:406)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:179)
		at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:566)
		... 36 more
	Caused by: java.lang.NoClassDefFoundError: jakarta/faces/context/FacesContext
		at java.base/java.lang.Class.getDeclaredFields0(Native Method)
		at java.base/java.lang.Class.privateGetDeclaredFields(Unknown Source)
		at java.base/java.lang.Class.getDeclaredFields(Unknown Source)
		at org.apache.catalina.startup.WebAnnotationSet.loadFieldsAnnotation(WebAnnotationSet.java:199)
		at org.apache.catalina.startup.WebAnnotationSet.loadApplicationListenerAnnotations(WebAnnotationSet.java:91)
		at org.apache.catalina.startup.WebAnnotationSet.loadApplicationAnnotations(WebAnnotationSet.java:68)
		at org.apache.catalina.startup.ContextConfig.applicationAnnotationsConfig(ContextConfig.java:316)
		at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:972)
		at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:290)
		at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:109)
		at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:4346)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:164)
		... 37 more
	Caused by: java.lang.ClassNotFoundException: jakarta.faces.context.FacesContext
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1228)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1059)
		... 49 more
```

### Are you interested in making a pull request?

None

> Caused by: java.lang.ClassNotFoundException: jakarta.faces.context.FacesContext

The cause is omnifaces 4 vs omnifaces 3. Apache Shiro depends on version 3.

#### Workaround
Most testing done with 5.5.0 but it should work for 5.4.0 as well.

In the unpacked war file, in `webapps/fuseki/WEB-INF/lib/`, change `omnifaces.4.x.x.jar` for [`omnifaces-3.14.9`](https://repo1.maven.org/maven2/org/omnifaces/omnifaces/3.14.9/).

#### Fuseki Evolution

Another solution is to switch to using the `apache-jena-fuseki` download and running the`fuseki-server.jar` self-contained jar file. This includes the UI.

It used to be the same code as the war file but since Jena 5.3.0, `fuseki-server.jar` is not based on the jakarta EE application any more.

Development of the webapp going into the war file is maintenance only now. war files are inflexible; the idea of running multiple applications in one server just isn't the way things are today. The project, like any independent open source project, does not have a lot of resources to keep everything going forever.

The project is moving to having the fuseki-server.jar file as the primary way of running Fuseki and getting new features. It seems to be the way most people run Fuseki and judging by the way the 5.4.0 war file fails, and not getting any reports that I can recall, the war file does not appear to get much use.

--- a/jena-fuseki2/jena-fuseki-webapp/pom.xml
+++ b/jena-fuseki2/jena-fuseki-webapp/pom.xml
@@ -89,18 +89,6 @@
       <groupId>org.apache.shiro</groupId>
       <artifactId>shiro-jakarta-ee</artifactId>
       <classifier>jakarta</classifier>
-      <exclusions>
-        <exclusion>
-          <groupId>org.omnifaces</groupId>
-          <artifactId>omnifaces</artifactId>
-        </exclusion>
-      </exclusions>
-    </dependency>
-    <!-- Include omnifaces -->
-    <dependency>
-      <groupId>org.omnifaces</groupId>
-      <artifactId>omnifaces</artifactId>
-      <version>${ver.omnifaces}</version>
     </dependency>
 
     <!-- frontend assets (HTML, JS, CSS, images, etc.) -->

--- a/pom.xml
+++ b/pom.xml
@@ -75,7 +75,6 @@
          See jena-fuseki-webapp and https://shiro.apache.org/jakarta-ee.html
     -->
     <ver.shiro>2.0.5</ver.shiro>
-    <ver.omnifaces>4.6.5</ver.omnifaces>
 
     <ver.protobuf>4.31.1</ver.protobuf>
     <ver.libthrift>0.22.0</ver.libthrift>

